pkg/zabbix_api.go

```diff
@@ -275,3 +280,336 @@ func isNotAuthorized(message string) bool {
 		message == "Not authorised." ||
 		message == "Not authorized."
 }
+
+func (ds *ZabbixDatasource) queryNumericItems(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+	jsonQueries := make([]*simplejson.Json, 0)
+	for _, query := range tsdbReq.Queries {
+		json, err := simplejson.NewJson([]byte(query.ModelJson))
+		if err != nil {
+			return nil, err
+		}
+
+		jsonQueries = append(jsonQueries, json)
+	}
+
+	if len(jsonQueries) == 0 {
+		return nil, errors.New("At least one query should be provided")
+	}
+
+	firstQuery := jsonQueries[0]
+
+	groupFilter := firstQuery.GetPath("group", "filter").MustString()
+	hostFilter := firstQuery.GetPath("host", "filter").MustString()
+	appFilter := firstQuery.GetPath("application", "filter").MustString()
+	itemFilter := firstQuery.GetPath("item", "filter").MustString()
+
+	items, err := ds.getItems(ctx, tsdbReq.GetDatasource(), groupFilter, hostFilter, appFilter, itemFilter, "num")
+	if err != nil {
+		return nil, err
+	}
+
+	metrics, err := ds.queryNumericDataForItems(ctx, tsdbReq, items, jsonQueries, isUseTrend(tsdbReq.GetTimeRange()))
+	if err != nil {
+		return nil, err
+	}
+
+	return BuildMetricsResponse(metrics)
+}
+
+func (ds *ZabbixDatasource) getItems(ctx context.Context, dsInfo *datasource.DatasourceInfo, groupFilter string, hostFilter string, appFilter string, itemFilter string, itemType string) (zabbix.Items, error) {
+	hosts, err := ds.getHosts(ctx, dsInfo, groupFilter, hostFilter)
+	if err != nil {
+		return nil, err
+	}
+	var hostids []string
+	for _, k := range hosts {
+		hostids = append(hostids, k["hostid"].(string))
+	}
+
+	apps, err := ds.getApps(ctx, dsInfo, groupFilter, hostFilter, appFilter)
+	if err != nil {
+		return nil, err
+	}
+	var appids []string
+	for _, l := range apps {
+		appids = append(appids, l["applicationid"].(string))
+	}
+
+	var allItems *simplejson.Json
+	if len(hostids) > 0 {
+		allItems, err = ds.getAllItems(ctx, dsInfo, hostids, nil, itemType)
+	} else if len(appids) > 0 {
+		allItems, err = ds.getAllItems(ctx, dsInfo, nil, appids, itemType)
+	}
+
+	itemsJSON, err := allItems.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var items zabbix.Items
+	err = json.Unmarshal(itemsJSON, &items)
+	if err != nil {
+		return nil, err
+	}
+
+	filteredItems := zabbix.Items{}
+	for _, item := range items {
+		if item.Status == "0" {
+			matched, err := regexp.MatchString(itemFilter, item.Name)
```

Comment:
This filter doesn't work. There're 2 cases in filters:
1. Filter is a group/host/app/item name. So exact string comparison should be used.
1. Filter is a regex (wrapped in `/`, like `/CPU (.*) time/`). Since go regex should be compiled without first and last `/`, it should be removed before `MatchString()` execution.

See https://github.com/alexanderzobnin/grafana-zabbix/blob/master/src/datasource-zabbix/zabbix/zabbix.js#L417
```js
function findByFilter(list, filter) {
  if (utils.isRegex(filter)) {
    return filterByRegex(list, filter);
  } else {
    return findByName(list, filter);
  }
}
```

Reply:
We're working on the Regex filter now. I pointed this out earlier but we set it aside to get a PR out before the holidays. The trickiest bit right now is handling flags, which a few are currently supported but the syntax for that is different in Go.

Reply:
I have updated the Regex filter the following way:
1) Check if the filter is a regex 
2) If it is a regex check for any flags, separate them and covert them to go regex format
3) If it is not regex do a regular string comparison

https://github.com/alexanderzobnin/grafana-zabbix/pull/847/commits/8bb98ae7a561048a307c4f13f214f4e39cd7d2e0

Reply:
Also b6c2116
Added unit tests and added a check for supported flags.